test/test_THDM_py_interface.py.in

```diff
@@ -0,0 +1,378 @@
+#!/usr/bin/env python
+
+from __future__ import print_function
+import os
+from decimal import Decimal
+
+import cppyy
+
+gm2calcpath = "@PROJECT_SOURCE_DIR@"
+# Add gm2calc headers to path
+includepath = os.path.join(gm2calcpath,"include")
+testpath = os.path.join(gm2calcpath,"test")
+cppyy.add_include_path(includepath)
+cppyy.add_include_path(testpath)
+    
+# Add Eigen3/Core directory to path
+eigen3path = "@EIGEN3_INCLUDE_DIR@"
+cppyy.add_include_path(eigen3path)
+
+cppyy.include(os.path.join("gm2calc","gm2_1loop.hpp"))
+cppyy.include(os.path.join("gm2calc","gm2_2loop.hpp"))
+cppyy.include(os.path.join("gm2calc","gm2_uncertainty.hpp"))
+cppyy.include(os.path.join("gm2calc","gm2_error.hpp"))
+cppyy.include(os.path.join("gm2calc","THDM.hpp"))
+cppyy.include(os.path.join("gm2calc","SM.hpp"))
+
+# Load library containing gm2calc definitions
+librarypath = "@CMAKE_LIBRARY_OUTPUT_DIRECTORY@"
+cppyy.add_library_path(librarypath)
+cppyy.load_library("libgm2calc")
+
+cppyy.cppdef("#define DOCTEST_CONFIG_IMPLEMENT_WITH_MAIN 1")
+
+cppyy.include("doctest.h")
+
+# Load data types
+from cppyy.gbl import std
+from cppyy.gbl import Eigen
+# Shorten class call name
+Matrix3cd = Eigen.Matrix[std.complex['double'],3,3]
+
+from cppyy.gbl import gm2calc
+from cppyy.gbl.gm2calc import SM
+from cppyy.gbl.gm2calc import THDM
+from cppyy.gbl.gm2calc import Error
+
+passed = 0
+errors = 0
+
+def testclose(val1,val2):
+    global passed, errors
+    try:
+        if (abs(Decimal(val1) - Decimal(val2)) == Decimal(0.)):
+            print("Error: expressions are not equal:",str(val1),"=!=",str(val2))
+            errors += 1
+        else:
+            passed += 1
+    except ValueError:
+        print("Non-numeric value found: val1 =",str(val1),"val2 =",str(val2),"eps =",str(eps))
```

Comment:
I think this error message should be changed "Values are not equal: " ...

Reply:
Hi Alex, no this error should be thrown if we try to pass a non-number into the function.  I have fixed it to match the correct error type.

Reply:
Ah, sorry, I didn't know this. :)